### ISSUE TITLE
ACORN Aggregation does not work for delayed mode

### ISSUE THREAD

**Issue by kereid** [aodn/aodn-portal#1692]
App version: 4.1.0
            Build date: Thu Apr 02 10:31:40 AEDT 2015
            Build number: 2
            Git ref: 66644d52db0a7f852d6a4f48d6725318390f16ce

Steps to reproduce:

Add IMOS - ACORN - Rottnest Shelf HF ocean radar site (Western Australia, Australia) - Delayed mode sea water velocity to the map change dates to be only 2 days apart (so not much data requested) proceed to step 3 and opt to download as netCDF

What should happen:

Download is successful

What does happen:

Download fails and get output:

Applying subset 'TIME,2014-06-01T00:30:00.000Z,2014-06-02T23:30:00.000999Z;LATITUDE,-29.8837890625,-13.0966796875;LONGITUDE,108.6767578125,120.9814453125'
Processing file 'IMOS_ACORN_V_20140602T003000Z_ROT_FV01_1-hour-avg.nc'
Processing file 'IMOS_ACORN_V_20140602T173000Z_ROT_FV01_1-hour-avg.nc'
Processing file 'IMOS_ACORN_V_20140601T023000Z_ROT_FV01_1-hour-avg.nc'
Processing file 'IMOS_ACORN_V_20140601T213000Z_ROT_FV01_1-hour-avg.nc'

@pblain do you want this up as a VIT? I did try real time ACORN and they worked.

**Comment by danfruehauf** [comment]
A bit more from the log:
```
ESC[32m[Thu Apr 2 13:32:32 AEDT 2015] INFO: Applying profile '/mnt/ebs/tomcat7/gogoduck/webapps/gogoduck/resources/worker/profiles/default'ESC[0m
ESC[32m[Thu Apr 2 13:32:32 AEDT 2015] INFO: Applying subset '-d LATITUDE,-29.8837890625,-13.0966796875 -d LONGITUDE,108.6767578125,120.9814453125' to 
'/tmp/tmp.AnhXN8BIRN/IMOS_ACORN_V_20140304T173000Z_COF_FV01_1-hour-avg.nc'ESC[0m
ESC[32m[Thu Apr 2 13:32:32 AEDT 2015] INFO: Applying subset '-d LATITUDE,-29.8837890625,-13.0966796875 -d LONGITUDE,108.6767578125,120.9814453125' to 
'/tmp/tmp.AnhXN8BIRN/IMOS_ACORN_V_20140304T143000Z_COF_FV01_1-hour-avg.nc'ESC[0m
ESC[32m[Thu Apr 2 13:32:32 AEDT 2015] INFO: Applying subset '-d LATITUDE,-29.8837890625,-13.0966796875 -d LONGITUDE,108.6767578125,120.9814453125' to 
'/tmp/tmp.AnhXN8BIRN/IMOS_ACORN_V_20140305T043000Z_COF_FV01_1-hour-avg.nc'ESC[0m
ESC[32m[Thu Apr 2 13:32:32 AEDT 2015] INFO: Applying subset '-d LATITUDE,-29.8837890625,-13.0966796875 -d LONGITUDE,108.6767578125,120.9814453125' to 
'/tmp/tmp.AnhXN8BIRN/IMOS_ACORN_V_20140304T023000Z_COF_FV01_1-hour-avg.nc'ESC[0m
ncks: ERROR User-specified coordinate value range 108.677 <= LONGITUDE <= 120.981 does not fall within valid coordinate range 153.011 <= LONGITUDE <= 
154.776
ncks: ERROR User-specified coordinate value range 108.677 <= LONGITUDE <= 120.981 does not fall within valid coordinate range 153.011 <= LONGITUDE <= 
154.776
ncks: ERROR User-specified coordinate value range 108.677 <= LONGITUDE <= 120.981 does not fall within valid coordinate range 153.011 <= LONGITUDE <= 
154.776
ncks: ERROR User-specified coordinate value range 108.677 <= LONGITUDE <= 120.981 does not fall within valid coordinate range 153.011 <= LONGITUDE <= 
154.776
ESC[33m[Thu Apr 2 13:32:32 AEDT 2015] WARN: Failed applying '-d LATITUDE,-29.8837890625,-13.0966796875 -d LONGITUDE,108.6767578125,120.9814453125' on 
file 'IMOS_ACORN_V_20140304T023000Z_COF_FV01_1-hour-avg.nc': ESC[0m
xargs: /bin/bash: exited with status 255; aborting
ESC[33m[Thu Apr 2 13:32:32 AEDT 2015] WARN: Failed applying '-d LATITUDE,-29.8837890625,-13.0966796875 -d LONGITUDE,108.6767578125,120.9814453125' on 
file 'IMOS_ACORN_V_20140304T173000Z_COF_FV01_1-hour-avg.nc': ESC[0m
xargs: /bin/bash: exited with status 255; aborting
```

I'm not sure yet what just happened there...

**Comment by kereid** [comment]
And it did work in production.

**Comment by kereid** [comment]
If bounding box does not intersect with any data on the map the download fails

**Comment by danfruehauf** [comment]
To clarify - this is a gogoduck bug. If **ANY** of the files returns an error (in this case, the bounding box does not intersect at all) - then the whole aggregation fails.

I'm still trying to figure out what's the best behaviour in this case.

**Comment by smancini** [comment]
I have just discussed this issue with Dan.

If the bounding box drawn by the user does not intersect at all with the extent of the radar site, then currently Gogoduck try to aggregate the different files but failed.
The aggregation is failing because there is no data in the area of interest.

The correct behaviour should be that we prevent the user to not download the data by turning in grey the "download" button on step 3. 
We should be able to do this by checking that the bounding box drawn by the user in step 2 intersects with the bounding box specified in the metadata for that collection.

**Comment by dnahodil** [comment]
I think we have everything in place to be able to achieve what @smancini describes without writing any new code in the Portal. The download button is not shown if we know there is no matching data. So if the data table (where we store the harvested file URLs) had a geometry column then we could match against that as we do for other collections. A bounding polygon would give us the best results. So I'd proposed a harvester update to fix this.

**Comment by danfruehauf** [comment]
The question is whether we want to have another column which will include exactly the same geometry (for every file in the collection). It sounds like a more generic approach, but in terms of modeling the data, this is also a redundancy. @ggalibert @lbesnard - what's your opinion?

The alternative doesn't sounds like too much code either on the portal side. But it's also a good question whether we want to add even more specific logic to gridded data (a.k.a `NcWmsPanel.js`).

**Comment by dnahodil** [comment]
What would you achieve by having a column which was the bounding area of the whole collection?

**Comment by danfruehauf** [comment]
So this will be solved when implementing:
https://github.com/aodn/backlog/issues/26

This is not quite a bug anymore but rather an improvement to the handling of gridded data in portal/GGD.

I added the required steps to the above backlog item. The end result will be better user experience and more consistent behaviour.

**Comment by danfruehauf** [comment]
After talking to @smancini this morning I learned that gridded data must **ALWAYS** have the same grid size - if there's no data in some areas it'll be filled with NULLs.

I suspected that some ACORN files may not correspond to the same grid. I tried to reproduce the original error @kereid had experienced but got to the conclusion that not just one file fails - but all of them. So things are behaving as they should and I'm more content now after verifying that.
